### ISSUE TITLE
Image assets don't work in custom widgets

### ISSUE THREAD

**Issue by abdrzq16** [FlutterFlow/flutterflow-issues#522]
**Expected behavior (required)**

Calling image from the assets in custom widget is not working at all. In all browsers and in the APK file.

**Current behavior (required)**

Just plain white.

**To Reproduce (required)**

If you need to use an Image. Asset in your custom code, it will not work.

```
 // Automatic FlutterFlow imports
import '../../flutter_flow/flutter_flow_theme.dart';
import '../../flutter_flow/flutter_flow_util.dart';
import '../widgets/index.dart'; // Imports other custom widgets
import '../../flutter_flow/custom_functions.dart'; // Imports custom functions
import 'package:flutter/material.dart';
// Begin custom widget code
// DO NOT REMOVE OR MODIFY THE CODE ABOVE!

class Aaa extends StatefulWidget {
  const Aaa({
    Key? key,
    this.width,
    this.height,
    this.ppp,
  }) : super(key: key);

  final double? width;
  final double? height;
  final String? ppp;

  @override
  _AaaState createState() => _AaaState();
}

class _AaaState extends State<Aaa> {
  @override
  Widget build(BuildContext context) {
    return Container(
      width: double.infinity,
      height: double.infinity,
      decoration: BoxDecoration(
        color: FlutterFlowTheme.of(context).secondaryBackground,
      ),
      child: 
        Image.asset(
                               'assets/images/604.jpg',
                                width: double.infinity,
                                height: double.infinity,
                                fit: BoxFit.contain,
                              ),

    );
  }
}

```
**Context (required)**

I had to make pageview of 600 images manually>

**Comment by organicnz** [comment]
Hi @abdrzq16. Thank you for adding this issue, mate 🙂

Apologies for asking, but could you pls add a bit more details? We need reproducible steps and screenshots, short video clips would be also very helpful :)

**Comment by abdrzq16** [comment]
I hope this is good for demonstrate this issue.

https://www.youtube.com/watch?v=FdOIcbKn3wE

**Comment by abdrzq16** [comment]
And I tried to use SVG in custom widgets both from assets and network using flutter_SVG, but I still get the same result.

**Comment by organicnz** [comment]
Thanks for the clarification, mate. Hmm, have you compiled this widget without errors? I can't get it to work for some reason :)

![image](https://user-images.githubusercontent.com/4765090/221423988-1d08dcd6-1166-4249-a229-cc4f6f187179.png)

**Comment by organicnz** [comment]
Ohh, I forgot to rename the custom widget. So, yeah, confirmation of the same result. Anyway, you can switch to the image element for now :)

**Comment by abdrzq16** [comment]
Thank you for that. And please don't forget to check SVG images too.

Regards.

**Comment by organicnz** [comment]
@abdrzq16 done. Apologies, I forgot to mention that on the left is the uploaded SVG asset, the second transparent in the grey area is SVG via the network, and then there are white blank spaces from custom widgets hehe

**Comment by wenkaifan0720** [comment]
Hi @abdrzq16. I just created a test project which can display asset images (https://app.flutterflow.io/project/flutterflow-issue-522-9wktd2). Could you share the bug report code? Thanks.

**Comment by abdrzq16** [comment]
As you can see the widget is compiled bat the image isn't working. So there is no bug report code.
![image](https://user-images.githubusercontent.com/108595330/221821350-fa176433-814f-4327-90fd-f97f1e8b2407.png)

**Comment by abdrzq16** [comment]
Well, I am sorry for this, but I have tried to use SVG from network and assets in the image element. It works fine in test mode, but when I build the APK from both the developer menu or the deployment page and install it, it doesn't work.
So I made a custom widget to add "flutter_svg: ^2.0.2" in the Dependencies, but still no result in the APK.

![image](https://user-images.githubusercontent.com/108595330/221916241-c836771b-ce57-4f9a-8de3-29c5bd105a20.png)

**Comment by wenkaifan0720** [comment]
Hi @abdrzq16. Using a asset image in a custom widget will not work when you only preview the custom widget itself. Because it does not have access to the assets in this condition. I have tested the full app on web and on Android and saw that the asset image can be shown. Is it not the case for you? If not, I still need you to provide the bug report code for me to inspect your project and see what is going wrong.

**Comment by abdrzq16** [comment]
When The image name was 'll3.WebP' it didn't work, but when I changed it to 'll-3.WebP' it worked.

Now, I have tried every way I could think of, but I couldn't make it work from variable.

This is the bur report code for this test project:
ITEez/Hqx89hn7wC0JP5YPowlQMwGkQ5a7sviu5+FBQaI+PuP4YEO+LeaklWYuqHTQpmOFGKmjgGpfD3vvLDJewpC0uabtlR/7oIUxDgQkiWVaq7DKm/PXdSM/NMGhGg4banhSRNBM1tLVoW21OUPfOua3qCf9qOYwx5e6fDbOY=

**Comment by wenkaifan0720** [comment]
Hi @abdrzq16. I somehow can not even download the asset images so it also didn't work for me. Have you tried using some other image format like png/jpg?

**Comment by abdrzq16** [comment]
Hi @wenkaifan0720 . Thank you for keeping up with me, and for pointing to the real issue here.

I think the problem is with codemagic. If the asset is not called from the code, it won't be in the APK. If it is a variable, codemagic won't expect what images to pack.

What I did is after I made the custom widget, I added image elements in another page with the same images I've chosen for the widget to force it to pack the images, and it worked for all image types.

So I suggest that FF team make an option to produce all the assets even if it not in the code, with a label to explain this situation.

And it will be better if they make it possible to make assets from variable in the image element itself.
